sujeong/app.js

```diff
@@ -50,6 +50,116 @@ app.post("/users", async (req, res) => {
   res.status(201).json({ message: "userCreated" });
 });
 
+app.post("/posts", async (req, res) => {
+  const { title, content, userId, postImageUrl } = req.body;
+
+  await appDataSource.query(
+    `INSERT INTO posts (
+      title,
+      content,
+      user_id,
+      image_url )
+      VALUES (?, ?, ?, ?) ;`,
+    [title, content, userId, postImageUrl]
+  );
+
+  res.status(201).json({ message: "postCreated" });
+});
+
+app.get("/search", async (req, res) => {
+  const search = await appDataSource.manager.query(
+    `SELECT
+    users.id AS userId,
+    users.profile_image AS userProfileImage, 
+    posts.id AS postingId,
+    posts.image_url AS postingImageUrl,
+    posts.content AS postingContent
+    FROM posts
+    INNER JOIN users ON posts.user_id=users.id`
+  );
+  res.status(200).json({ data: search });
+});
+
+app.get("/users/:userId/posts", async (req, res) => {
+  const { userId } = req.params;
+  const result = await appDataSource.query(
+    `SELECT
+    users.id AS userId,
+    users.profile_image AS userProfileImage, 
+    JSON_ARRAYAGG(
+    JSON_OBJECT(
+    "postingId", posts.id,
+    "postingImageUrl", posts.image_url,
+    "postingContent", posts.content ))
+    AS postings
+    FROM users
+    JOIN posts ON posts.user_id=users.id
+    WHERE
+    users.id=?
+    GROUP BY users.id
+  `,
+    [userId]
+  );
+
+  return res.status(200).json({ data: result[0] });
+});
+
+app.patch("/posting_update/:postId", async (req, res) => {
+  const { postId } = req.params;
+  const { postingContent } = req.body;
+
+  await appDataSource.query(
+    `UPDATE posts
+    SET
+    content=?
+    WHERE id=?
+    `,
+    [postingContent, postId]
+  );
+
+  const update = await appDataSource.query(
+    `SELECT
+      u.id AS userId,
+      u.name AS userName,
+      p.id AS postingId,
+      p.title AS postingTitle,
+      p.content AS postingContent
+      FROM posts AS p
+      JOIN users AS u ON p.user_id=u.id
+      WHERE p.id=?
+      `,
+    [postId]
+  );
+  res.status(201).json({ data: update[0] });
+});
```

Comment:
1. 변수명은 최대한 어떠한 데이터가 담겨있는지를 표시해주는 것이 좋습니다.
2. post라는 변수를 배열로 감쌌는데, 이렇게하면 post라는 변수에 결과로 올 배열의 첫번째 값이 post 변수에 담깁닌다. 그래서 마지막에 post[0]처럼 인데스를 안 사용할 수 있습니다.
3. status code 201은 데이터가 생성 될 때 사용하므로 200으로 수정해 주면 좋을 것 같습니다.
```suggestion
  const [post] = await appDataSource.query(
    `SELECT
      u.id AS userId,
      u.name AS userName,
      p.id AS postingId,
      p.title AS postingTitle,
      p.content AS postingContent
      FROM posts AS p
      JOIN users AS u ON p.user_id=u.id
      WHERE p.id=?
      `,
    [postId]
  );
  res.status(200).json({ data:post });
});
```